packages/generate-persisted-query-manifest/src/__tests__/cli.test.ts

```diff
@@ -0,0 +1,1105 @@
+    ${query}
+    ${fragment}
+  `;
+
+  expect(code).toBe(0);
+  expect(manifest).toBeManifestWithOperations([
+    {
+      id: sha256(addTypenameToDocument(combined)),
+      name: "CurrentUserQuery",
+      body: print(addTypenameToDocument(combined)),
+      type: "query",
+    },
+  ]);
+
+  await cleanup();
+});
+
+test("handles interpolated fragments in queries used with `gql`", async () => {
+  const { cleanup, writeFile, readFile, runCommand } = await setup();
+  const query = gql`
+    query CurrentUserQuery {
+      currentUser {
+        ...CurrentUserFragment
+      }
+    }
+  `;
+  const fragment = gql`
+    fragment CurrentUserFragment on CurrentUser {
+      id
+    }
+  `;
+
+  await writeFile(
+    "./src/my-component.js",
+    `
+import { gql } from '@apollo/client';
+import { fragment } from './fragment';
+
+const query = gql\`
+  query CurrentUserQuery {
+    currentUser {
+      ...CurrentUserFragment
+    }
+  }
+
+  \${fragment}
+\`
+`,
+  );
+  await writeFile(
+    "./src/fragment.js",
+    `
+import { gql } from '@apollo/client';
+
+export const fragment = gql\`
+  fragment CurrentUserFragment on CurrentUser {
+    id
+  }
+\`;
+`,
+  );
+
+  const { code } = await runCommand();
+
+  const manifest = await readFile("./persisted-query-manifest.json");
+  const combined = addTypenameToDocument(gql`
+    ${query}
+    ${fragment}
+  `);
+
+  expect(code).toBe(0);
+  expect(manifest).toBeManifestWithOperations([
+    {
+      id: sha256(addTypenameToDocument(combined)),
+      name: "CurrentUserQuery",
+      body: print(combined),
+      type: "query",
+    },
+  ]);
+
+  await cleanup();
+});
+
+test("handles implied fragments in queries from `gql`", async () => {
+  const { cleanup, writeFile, readFile, runCommand } = await setup();
+  const query = gql`
+    query CurrentUserQuery {
+      currentUser {
+        ...CurrentUserFragment
+      }
+    }
+  `;
+  const fragment = gql`
+    fragment CurrentUserFragment on CurrentUser {
+      id
+    }
+  `;
+
+  await writeFile(
+    "./src/my-component.js",
+    `
+import { gql } from '@apollo/client';
+
+const query = gql\`
+  query CurrentUserQuery {
+    currentUser {
+      ...CurrentUserFragment
+    }
+  }
+\`
+`,
+  );
+  await writeFile(
+    "./src/fragment.js",
+    `
+import { gql } from '@apollo/client';
+
+export const fragment = gql\`
+  fragment CurrentUserFragment on CurrentUser {
+    id
+  }
+\`;
+`,
+  );
+
+  const { code } = await runCommand();
+
+  const manifest = await readFile("./persisted-query-manifest.json");
+  const combined = addTypenameToDocument(gql`
+    ${query}
+    ${fragment}
+  `);
+
+  expect(code).toBe(0);
+  expect(manifest).toBeManifestWithOperations([
+    {
+      id: sha256(addTypenameToDocument(combined)),
+      name: "CurrentUserQuery",
+      body: print(combined),
+      type: "query",
+    },
+  ]);
+
+  await cleanup();
+});
+
+test("writes manifest file with no operations when none found", async () => {
```

Comment:
isn't this the same as the first test?

Reply:
Technically yes, but with the added check for the empty list of operations, not just the existence of the file. The only reason I separated this was to call out this behavior explicitly (i.e. "no operations" as the precondition) rather than making one long test description. Thoughts?

Reply:
Will merge for now, but if you feel strongly that this should be combined, let me know and I can put this in a followup PR. Thanks!